### ISSUE TITLE
document "--no-XXX" handling

### ISSUE THREAD

**Issue by getify** [substack/minimist#54]
I had to read the source code to diagnose why `--no-foobar` was not working as I expected (it wasn't giving me a `"--no-foobar": true` entry in my arguments. It was instead setting `"foobar": false`. Interesting feature, but should be documented.

**Comment by drifkin** [comment]
I'd like to see this documented as well, it's a nice feature, but a little confusing at first.

**Comment by bucaran** [comment]
Nice feature indeed. I usually call this _no-flags_ and it's common in many other cli parsers, commander, nopt, etc.

**Comment by lipis** [comment]
Does anyone know if it's possible to not change the argument name and keep it like `"--no-foobar": true` instead?! maybe some sort of option..?

**Comment by bucaran** [comment]
@lipis Hmm I don't think so. This is supposed to be a feature. What you could do is fork minimist and remove the --no- logic or try to adjust your application to use it instead.

**Comment by lipis** [comment]
yes.. that path is known and dangerous.. because it is going to be even more dead than this project.. :)

**Comment by bucaran** [comment]
@lipis This project _is_ like an old forgotten tree that keeps providing shade and shelter. It's not perfect, but it works and it's quite widespread. There are alternatives too, but minimist is often just good enough.

**Comment by lipis** [comment]
That is also true.. but I can't include my own branch to another open source project :) If there is no option to disable that.. that's all I needed to know. Thanks @bucaran

**Comment by lipis** [comment]
But on the side note.. I think @substack should at least reply to the #83

**Comment by bucaran** [comment]
@lipis TBH I would prefer if it didn't handle --no- options. I wrote a library, [parsec](https://github.com/bucaran/parsec), that handles --no flags too, but not because I had a real use case, but because other cli parsers like minimist did it too.

I wish someone would write a spec of what they think the perfect cli parser should look like and work like.

**Comment by lipis** [comment]
In my case I want to parse the arguments via gulp to pass them to another cli.. so changing the name of the argument is not going to work..

**Comment by aureooms** [comment]
I was expecting this feature but could not find it in the docs. Had to look up issues. Related #123.
